content/blog/callpackage-a-tool-for-the-lazy.md

```diff
@@ -0,0 +1,219 @@
+---
+title: callPackage, a tool for the lazy
+date: 2022-08-25
+extra:
+  author: Norbert Melzer
+---
+
+In [`nixpkgs`](https://github.com/nixos/nixpkgs), there is a massive use of the
+`callPackage` function, which provides us with a lot of benefits.
+
+## Basic examples
+
+Though before even discussing the benefits, lets see how it actually gets used.
+
+Given are these 2 files `hello.nix` and `default.nix`:
+
+```nix
+# Usually I would prefer the usage of nix flakes, though that would
+# introduce a lot of boilerplate in the examples for no reason,
+# therefore I will use diamond paths and `import` for simplicity.
+```
+
+```nix
+# default.nix
+let pkgs = import <nixpkgs> {}; in
+pkgs.callPackage ./hello.nix {}
+```
+
+```nix
+# hello.nix
+{writeShellScriptBin}:
```

Comment:
[`nixpkgs` style](https://nixos.org/manual/nixpkgs/stable/#chap-conventions) (sorry, no anchor to the relevant entry)

```suggestion
{ writeShellScriptBin }:
```